src/cache.ts

```diff
@@ -180,6 +180,67 @@ export class CMakeCache {
     return entries;
   }
 
+  async replace(content: string, key: string, value: boolean): Promise<string> {
+    return new Promise(resolve => {
+      const re = key + ':.+=(.+)';
+      const found = content.match(re);
+
+      if (found && found.length >= 2) {
+        const line = found[0];
+        const currentVal = found[1];
+        const newValueLine = line.replace(currentVal, value ? 'TRUE' : 'FALSE');
+        resolve(content.replace(line, newValueLine));
+      } else {
+        resolve('');
+      }
+    });
+  }
+
+  async replaceOption(key: string, value: boolean) {
+    return new Promise(async resolve => {
+      const exists = await fs.exists(this.path);
```

Comment:
It is unclear to me why you need to check the existence of the file and read it again.  I would have expected the cache entries to have already been parsed and placed in the `this._entries` member variable.

Reply:
(this feedback applies to all new functions added to this class)

Reply:
will any changes done to the `this._entries` variable be persisted? @bobbrow